Update mtalua.tmLanguage.json

Loki214 opened the pull request:
Most of the functions were already there and some were in the wrong places.

Loki214 committed:
Update mtalua.tmLanguage.json

Most of the functions were already there and some were in the wrong places.

Subtixx commented:
Is this still needed? I only checked Client & Server against the newest and find no differences except that you still have the deprecated functions in there?

Also, you've to consider all those hundreds of functions were not typed out by myself. I would have gone crazy if I did so. Rather I wrote a scraper (also found in this repository) which scrapes the MTA Wiki for me for those functions + descriptions

Loki214 committed:
Update mtalua.tmLanguage.json

These were moved to the shared matches.
Vector2, Vector3, Vector4, Matrix, root, resourceRoot, debugSleep, getDevelopmentMode, setDevelopmentModeengineLoadIFP, 

Also the new animation function were added to client matches
engineReplaceAnimation, engineRestoreAnimation

Loki214 commented:
Yeah you should merge it now I just fixed some duplicates I made
also the changes were 
|Vector2|Vector3|Vector4|Matrix|root|resourceRoot|debugSleep|getDevelopmentMode|setDevelopmentMode
were moved to shared matches

engineLoadIFP|engineReplaceAnimation|engineRestoreAnimation
were added to the client matches

Subtixx commented on syntaxes/mtalua.tmLanguage.json:
```diff
@@ -236,7 +236,7 @@
 		},
 		{
-			"match": "(?<![^.]\\.|:)\\b(coroutine\\.(create|resume|running|status|wrap|yield)|string\\.(byte|char|dump|find|format|gmatch|gsub|len|lower|match|rep|reverse|sub|upper)|table\\.(concat|insert|maxn|remove|sort)|math\\.(abs|acos|asin|atan2?|ceil|cosh?|deg|exp|floor|fmod|frexp|ldexp|log|log10|max|min|modf|pow|rad|random|randomseed|sinh?|sqrt|tanh?)|io\\.(close|flush|input|lines|open|output|popen ...
+			"match": "(?<![^.]\\.|:)\\b(coroutine\\.(create|resume|running|status|wrap|yield)|string\\.(byte|char|dump|find|format|gmatch|gsub|len|lower|match|rep|reverse|sub|upper)|table\\.(concat|foreach|insert|maxn|remove|sort)|math\\.(abs|acos|asin|atan2?|ceil|cosh?|deg|exp|floor|fmod|frexp|ldexp|log|log10|max|min|modf|pow|rad|random|randomseed|sinh?|sqrt|tanh?)|io\\.(close|flush|input|lines|open|outp ...
```
This is the only change that was not brought in with the changes today. But I wont merge all of your stuff just because of table.foreach

Subtixx commented on syntaxes/mtalua.tmLanguage.json:
```diff
@@ -208,15 +208,15 @@
 		},
 		{
-			"match": "\\b((addAccount|copyAccountData|getAccountData|getAccountName|getAccountPlayer|getAccountsBySerial|getAccountSerial|getAccounts|getAccount|getAllAccountData|getPlayerAccount|isGuestAccount|logIn|logOut|removeAccount|setAccountData|setAccountPassword|aclReload|aclSave|aclList|aclGet|aclCreate|aclDestroy|aclGetName|aclGetRight|aclSetRight|aclListRights|aclRemoveRight|aclGetGroup|aclCre ...
+			"match": "\\b((addAccount|copyAccountData|getAccountData|getAccountName|getAccountPlayer|getAccountsBySerial|getAccountSerial|getAccounts|getAccount|getAllAccountData|getPlayerAccount|isGuestAccount|logIn|logOut|removeAccount|setAccountData|setAccountPassword|aclReload|aclSave|aclList|aclGet|aclCreate|aclDestroy|aclGetName|aclGetRight|aclSetRight|aclListRights|aclRemoveRight|aclGetGroup|aclCre ...
```
Missing:
```
dbPrepareString
resendPlayerACInfo
getCommandHandlers
isResourceArchived
```

And obsolete/deprecated:
```
isPedFrozen
setPedFrozen
```

Subtixx replied:
By the way that's the only differences I found with master vs this one.

Subtixx commented on syntaxes/mtalua.tmLanguage.json:
```diff
@@ -208,15 +208,15 @@
 		},
 		{
-			"match": "\\b(getRadioChannel|getRadioChannelName|getSFXStatus|getSoundBPM|getSoundEffects|getSoundFFTData|getSoundLength|getSoundLevelData|getSoundMaxDistance|getSoundMetaTags|getSoundMinDistance|getSoundPan|getSoundPosition|getSoundProperties|getSoundSpeed|getSoundVolume|getSoundWaveData|isSoundPanningEnabled|isSoundPaused|playSFX|playSFX3D|playSound|playSound3D|setRadioChannel|setSoundEffec ...
+			"match": "\\b(getRadioChannel|getRadioChannelName|getSFXStatus|getSoundBPM|getSoundEffects|getSoundFFTData|getSoundLength|getSoundLevelData|getSoundMaxDistance|getSoundMetaTags|getSoundMinDistance|getSoundPan|getSoundPosition|getSoundProperties|getSoundSpeed|getSoundVolume|getSoundWaveData|isSoundPanningEnabled|isSoundPaused|playSFX|playSFX3D|playSound|playSound3D|setRadioChannel|setSoundEffec ...
```
Only difference is that on my end
```
getPedSkin
```
is now removed in master.

Subtixx commented on syntaxes/mtalua.tmLanguage.json:
```diff
@@ -208,15 +208,15 @@
 		},
 		{
-			"match": "\\b(playSoundFrontEnd|createBlip|createBlipAttachedTo|getBlipColor|getBlipIcon|getBlipSize|setBlipColor|setBlipIcon|setBlipSize|getBlipOrdering|setBlipOrdering|getBlipVisibleDistance|setBlipVisibleDistance|fadeCamera|getCameraInterior|getCameraMatrix|getCameraTarget|setCameraInterior|setCameraMatrix|setCameraTarget|getBodyPartName|getClothesByTypeIndex|getClothesTypeName|getTypeIndex ...
+			"match": "\\b(playSoundFrontEnd|createBlip|createBlipAttachedTo|getBlipColor|getBlipIcon|getBlipSize|setBlipColor|setBlipIcon|setBlipSize|getBlipOrdering|setBlipOrdering|getBlipVisibleDistance|setBlipVisibleDistance|fadeCamera|getCameraInterior|getCameraMatrix|getCameraTarget|setCameraInterior|setCameraMatrix|setCameraTarget|getBodyPartName|getClothesByTypeIndex|getClothesTypeName|getTypeIndex ...
```
Only difference between master and this is:

master is missing
```
showPlayerHudComponent
```
now.

Subtixx commented:
Sorry but I don't see the point in merging this. I highlighted the lines and explained why I think those changes aren't necessary anymore.

Subtixx reviewed (commented):
Sorry but I don't see the point in merging this. I highlighted the lines and explained why I think those changes aren't necessary anymore.

Loki214 commented:
Okay sure no prob
